### PR TITLE
Introduce PS_TRUSTED_PROXIES env, and delete the obsolete PS_SSL_ENABLED_EVERYWHERE config

### LINKED ISSUE #35487: Symfony Layout with the nightly docker image doesn't work
**Issue encounter**
I have not yet discovered the reason behind the loop redirect issue when accessing the BO. This problem seems to be present in the Docker version of PrestaShop from a Nightly image when the Symphony Layout is enabled.

**Expected behavior**
Nightly docker image should work with the Symphony Layout enabled.

**Acceptance test**
Docker nightly should work

### PR DESCRIPTION
**Pull request by M0rgan01**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Introduce `PS_TRUSTED_PROXIES` env allowing you to configure the authorized IPs if Prestashop is used behind a load-balancer or a reverse-proxy (see https://symfony.com/doc/current/deployment/proxies.html). <br/> Delete the obsolete `PS_SSL_ENABLED_EVERYWHERE` config, which allowed you to have SSL activated everywhere, partly becoming obsolete over time. `PS_SSL_ENABLED` initially allowed to have SSL activated **on customer account identification and order processing**.
| Type?             | improvement
| Category?         | BO
| BC breaks?        | yes
| Deprecations?     | no
| How to test?      | See  https://github.com/PrestaShop/PrestaShop/issues/35487. Use a reverse-proxy, configure the environment variable `PS_TRUSTED_PROXIES`, activate `PS_SSL_ENABLED` and checked the correct functioning of the application
| UI Tests          | https://github.com/M0rgan01/ga.tests.ui.pr/actions/runs/8455000992
| Fixed issue or discussion?     | Fixes #35487
| Related PRs       | https://github.com/PrestaShop/autoupgrade/pull/684
| Sponsor company   | -

### BC break

Remove the `PS_SSL_ENABLED_EVERYWHERE` config, use `PS_SSL_ENABLED` instead

### CODE REVIEW

**Review by jolelievre — COMMENTED** [review]
Thanks @M0rgan01 

all good for what you modifed, I think you also need to update the `configuration.xml` fixture file to remove this everywhere config from the installed data

Also this is a breaking change, although acceptable, so the PR description should indicate it And you'll need to update the autoupgrade scripts to remove the configuration so at least add the Autoupgrade label to avoid forgetting it

**Review by jolelievre — DISMISSED** [review]
Thanks @M0rgan01

**Review by tleon — DISMISSED** [review]

**Review by boherm — DISMISSED** [review]

File: install-dev/data/xml/configuration.xml [hunk 1]
```diff
@@ -86,9 +86,6 @@
   <configuration id="PS_NB_DAYS_NEW_PRODUCT" name="PS_NB_DAYS_NEW_PRODUCT">
     <value>20</value>
   </configuration>
-  <configuration id="PS_SSL_ENABLED" name="PS_SSL_ENABLED">
-    <value>0</value>
```
Comment (PrestaEdit) [inline install-dev/data/xml/configuration.xml]: Doesn't you need to keep it but change it's value if you want to get the 1 as default one ?

Comment (M0rgan01) [reply install-dev/data/xml/configuration.xml]: Thanks, that's a mistake. It is `PS_SSL_ENABLED_EVERYWHERE` which should have been removed but it is not present in this file.

**Review by jolelievre — APPROVED** [review]

**Review by matks — APPROVED** [review]

**Review by boherm — APPROVED** [review]

**Comment by boherm** [comment]
QA by dev done!
It's ok for me!

**Pull request merged**